feat(tabs): initial design guidance (#35)
vmware-clarity/clarity-guidance: 1032-tabs-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-18
updated: 2024-10-22
---

# Tabs Design Guidance

[Tabs](https://clarity.design/documentation/tabs) are components that organize the content of a page into multiple panes. Users can see the content of one tab at a time, yet quickly navigate to another tab.

## Guidance

> - **Must** place tabs in a single, non-scrollable row, above their content.

Reviewer: ```suggestion
- Design teams **must** place tabs in a single, non-scrollable row, positioned above their content to ensure clarity and ease of navigation.
```

> - **Must not** use tabs in cards and modals.

Reviewer: ```suggestion
- Design teams must not use tabs in cards or modals, as this can create clutter and reduce usability within constrained spaces.
```

> - **Must not** use more than seven tabs and limit labels to one or two words to ensure that all tabs appear in the container. **May** use [tab overflow](https://clarity.design/documentation/tabs/code#overflow-tabs) when the tab group is larger than its containing space or if you need to preserve space.

Reviewer: ```suggestion
- Design teams **must not** use more than seven tabs and must limit labels to one or two words to ensure all tabs fit within the container.  Design teams **may** use [tab overflow](https://clarity.design/documentation/tabs/code#overflow-tabs) when the tab group exceeds the available space or when space needs to be preserved.
```

> - **Must** avoid using more than seven tabs and limit labels to one or two words to ensure that all tabs appear in the container. You may use [Tab Overflow](https://clarity.design/documentation/tabs/code#overflow-tabs) when the tab group is larger than its containing space or if you need to preserve space.

Reviewer 2: ```suggestion
- **Must** avoid using more than seven tabs and limit labels to one or two words to ensure that all tabs appear in the container. You **may** use [Tab Overflow](https://clarity.design/documentation/tabs/code#overflow-tabs) when the tab group is larger than its containing space or if you need to preserve space.
```

"May" should bolded. Also, please update the "must avoid" based on my other comment.

> - **Must** avoid using more than seven tabs and limit labels to one or two words to ensure that all tabs appear in the container. You may use [Tab Overflow](https://clarity.design/documentation/tabs/code#overflow-tabs) when the tab group is larger than its containing space or if you need to preserve space.

Reviewer 2: ```suggestion
- **Must** avoid using more than seven tabs and limit labels to one or two words to ensure that all tabs appear in the container. You may use [tab overflow](https://clarity.design/documentation/tabs/code#overflow-tabs) when the tab group is larger than its containing space or if you need to preserve space.
```

I don't think "tab overflow" should be capitalized here.

> - **Must not** use tabs to break user interactions into a series of steps. Serial workflows are best presented in a [stepper](https://clarity.design/documentation/stepper) or [wizard](https://clarity.design/documentation/wizard). 

Reviewer: ```suggestion
- Design teams must not use tabs to break user interactions into a series of steps. Serial workflows are better presented using a [stepper](https://clarity.design/documentation/stepper) or a [wizard](https://clarity.design/documentation/wizard). 
```

> - **Must** NOT use tabs to break user interactions into a series of steps. Serial workflows are best presented in a [Stepper](https://clarity.design/documentation/stepper) or [Wizard](https://clarity.design/documentation/wizard). 

Reviewer 2: ```suggestion
- **Must not** use tabs to break user interactions into a series of steps. Serial workflows are best presented in a [Stepper](https://clarity.design/documentation/stepper) or [Wizard](https://clarity.design/documentation/wizard). 
```

We should change this in all files. This aligns with Google's AIP language: https://google.aip.dev/1.

> - **Must not** use tabs to break user interactions into a series of steps. Serial workflows are best presented in a [Stepper](https://clarity.design/documentation/stepper) or [Wizard](https://clarity.design/documentation/wizard). 

Reviewer 2: ```suggestion
- **Must not** use tabs to break user interactions into a series of steps. Serial workflows are best presented in a [stepper](https://clarity.design/documentation/stepper) or [wizard](https://clarity.design/documentation/wizard). 
```

These aren't proper nouns. I don't think we use capitalization.

> - **Should** use tabs for alternate views within a vertical nav or main content area.

Reviewer: ```suggestion
- Design teams **should** use tabs for alternate views within a vertical navigation or main content area to help users switch between related content efficiently.
```

> - **Should** use [vertical tabs](https://clarity.design/documentation/tabs/code#vertical-tabs) for vertical alignment.

Reviewer: ```suggestion
- Design teams **should** use [vertical tabs](https://clarity.design/documentation/tabs/code#vertical-tabs) when tabs need to align along the side of the content for better structure in narrow layouts.
```

> - **Should** use [Vertical Tabs](https://clarity.design/documentation/tabs/code#vertical-tabs) for vertical alignment.

Reviewer 2: ```suggestion
- **Should** use [vertical tabs](https://clarity.design/documentation/tabs/code#vertical-tabs) for vertical alignment.
```

Same here. Not a proper noun.

> - **Should** use [Vertical Tabs](https://clarity.design/documentation/tabs/code#vertical-tabs) for vertical alignment.

Author: Guidance is if designer needs vertical alignment need to use vertical tabs. It depends on case to case we can't specify situation when to use ...  vertical alignment is call of designer.

## Content Guidelines

> - **Must not** have cross-linking between tabs.

Reviewer: ```suggestion
- Design teams **must not** allow cross-linking between tabs, as it can confuse users and disrupt the intended navigation flow.
```

> - Content in each tab **should** be independent of the content in other tabs.

Reviewer: ```suggestion
- Content in each tab **should** be independent of the content in other tabs to prevent confusion and ensure a clear, focused user experience.
```

> - **Should** keep similar content in the same view. Don't force users to navigate back and forth to compare data.

Reviewer: ```suggestion
- Design teams **should** keep similar content within the same view to prevent unnecessary back-and-forth navigation, allowing users to compare data more easily and stay focused.
```

> - **May** consider dividing long content into subsections within the tab.

Reviewer: ```suggestion
- Design teams **may** consider dividing long content into subsections within a tab to improve readability and help users quickly find relevant information.
```

## Label Guidelines

> - **Must not** use generic labels such as General or Advanced.

Reviewer: ```suggestion
- Design teams **must not** use generic labels like "General" or "Advanced," as they provide little clarity and can confuse users about the content within the tabs.
```

> - **Must** use title case capitalization.

Reviewer: ```suggestion
- Design teams **must** use title case capitalization for tab labels to maintain consistency and improve readability.
```

> - **Must** Use title-style capitalization.

Reviewer 2: ```suggestion
- **Must** use title case for capitalization.
```

> - **Must not** use icons in labels.

Reviewer: ```suggestion
- Design teams **must not** use icons in tab labels. Icons can reduce clarity and increase cognitive load, making it harder for users to understand the content at a glance.
```

> - **Must** show a clear relationship between the tabs.

Reviewer: ```suggestion
- Design teams **must** show a clear relationship between the tabs to help users understand how the content is connected and navigate intuitively.
```

> - **Should** favor nouns over verbs, for example, Settings, Permissions, and Performance.

Reviewer: ```suggestion
- Design teams should favor nouns over verbs in tab labels, for example, "Settings," "Permissions," and "Performance," to create clearer, more scannable navigation.
```
